### PR TITLE
Move other query parameter usages to util - Fixes #1664

### LINKED ISSUE #1664: Add all query params to QueryParams class
The QueryParams class was added as a helper class for null safety, but we could refactor the code to use it for all of the query params.

Context: https://github.com/dart-lang/dart-pad/pull/1645/files/1d174687ad1a04cb667fd3d193ba569e829991cb#r515426045

### PR DESCRIPTION
**Pull request by parlough**
Fixes #1664

### CODE REVIEW

File: lib/util/query_params.dart [hunk 1]
```diff
@@ -1,20 +1,17 @@
 import 'dart:html';
 
-class QueryParams {
+/// A utility for accessing and setting query parameters.
+abstract class QueryParams {
```
Comment (parlough) [inline lib/util/query_params.dart:4]: I made this `abstract` so IDEs don't think I'm constantly trying to instantiate it instead of refer to its static members.

Comment (kevmoo) [inline lib/util/query_params.dart:4]: drive-by-comment.

Maybe better to make this a final, singleton, top-level field that instantiated on first use?

It's global state either way, but it feels better.

Comment (parlough) [reply lib/util/query_params.dart]: Thanks for the suggestion, that definitely makes sense :). I wasn't sure the most appropriate way to implement this change though, so if you wouldn't mind taking another look, that would be great.

Comment (kevmoo) [reply lib/util/query_params.dart]: Love it!

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
PTAL @johnpryan

**Comment by domesticmouse** [comment]
@parlough looks like there is a linter warning that needs addressing. Thanks!

**Comment by parlough** [comment]
@domesticmouse I rebased on master and addressed the lint. Let me know if you'd like to see any further changes :)

**Comment by domesticmouse** [comment]
LGTM. I'll give @johnpryan a couple of days to complain before I land this PR, force majeure style =)

**Review by johnpryan — APPROVED** [review]
no complaints, this looks good to me!
